### DISCUSSION TITLE
GitHub Preview tab of long/complex comment falsely remains filled after submission

### DISCUSSION THREAD

**Discussion by porg** [community/community#57607] [Projects and Issues]
### Select Topic Area

Bug

### Select Feature Area

- [X] Issues (The feature that allows you to track ideas, feedback, tasks, or bugs for work on GitHub. Not for any issue you are having).
- [ ] Projects (For planning and tracking work on GitHub - not any open source or other type of project).

## ✅ GitHub Preview tab of short/simple comment correctly clears after submission

https://github.com/community/community/assets/737143/3f85ebb1-e921-4575-a7f7-bb5207c9de85

## ❌  GitHub Preview tab of long/complex comment falsely remains filled after submission

https://github.com/community/community/assets/737143/e8279ea4-8e21-4a7e-ba64-3c90e3304ce2

## Reproduction

1. Draft a comment to an existing issue.
2. Write some markdown text in the "Write" tab.
    - The content makes a difference whether the bug manifests or not!
    - With shorter/simpler text the bug does not occur.
    - With longer or more complex markup (e.g. various headings, some emojis, some code blocks or inline code) this happens, e.g. with this [Sample comment](https://github.com/WordPress/gutenberg/issues/50327#issuecomment-1584257063).
3. Switch to the "Preview" tab.
4. Click the button to submit your issue or comment.
5. The content gets submitted indeed.
6. But due to the bug you may not notice that submission took place:

**Expected:** Both the "Write" and the "Preview" tabs shall reliably get their content cleared after submission, to signal to the user "content received, now the comment-widget is cleared and ready for another comment if needed".

**Actual:** The "Preview" tab still contains the content, which in reality already got submitted, and is just "a ghost from the past" which remained.
- It is not unlikely that the user is in "Preview" mode prior submitting, so this certainly needs to be supported.
- If your comment-panel filled the viewport-height (fully or to a large degree) you do not even see the history above, and hence do not even realize that the "Preview" tab is a "ghost representation which was not cleared properly".
- If the user still sees this s/he may falsely thinks that submission has failed.
- When switching to the "Write" tab it is empty When you switch back to the "Preview" panel again then it is empty too, so the clearing mechanism has kicked in then. Then you realize this was just a glitch.
- Nevertheless, the first few times, even I as a tech savvy user was confused/disoriented.
- Happened numerous times over the last months on GitHub.

## Environment
- Safari 15.6 on macOS 11.7.6 Big Sur

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by tokiory** [comment]
bump

**Comment by porg** [comment]
## Seems like a relatively small effort to fix quite a large usability bug

- Seemingly small technical glitch: The preview tab is not reset under all conditions.
  - And the precondition for the bug seems to be rather complex markup.
- My guts feeling in my capability as a software tester:
  - Possibly the rendering (partially client/server side) of more complex content in these situations is not yet finished and hence blocks other activities such as the tab reset, and as the submission takes place before that, the reset never occurs.
  - So the solution then would be that if submission takes place that the preview tab reset takes place with utmost important/priority, and cannot be blocked by other pending processes.

**Comment by porg** [comment]
@Github: Few upvotes doesn't mean no-one is bothered by this!
- Even though that the target audience on GitHub is quite techy: This is quite a complex bug which possibly not too many users can articulate on or assume the root cause, but which nevertheless puzzles and annoys them every here and there.
- Don't underestimate the annoyance / importance of this bug.

**Comment by porg** [comment]
Meanwhile the comment widget was redesigned
- ✅ Very minimalist (only a single pixel border, no "flesh/body at all) I like it overall!
- ❌ But stills the bug persists! (retested today with the same Markdown content in the meanwhile newer Safari 16.6.1)

https://github.com/community/community/assets/737143/1e78a8b8-1544-46d4-b68a-834d9479c50b

**Comment by porg** [comment]
👍 As of 2025-01 this was finally fixed!
- The editor panel in general was repolished in terms of padding and focus-color
- And the lifecycle now works correctly: When submitted both the "Write" (=editor) and "Preview" tab are cleaned.
- Also that it jumps to the submitted content top-aligned works perfectly now.
- The whole comment UX now is great.

Screen recording:

https://github.com/user-attachments/assets/a53e1840-36fb-4255-91f6-44fc02037321
